pkg/ovsdb/ovsdb.go

```diff
@@ -381,3 +333,53 @@ func (s *ServOVSDB) Echo(ctx context.Context, param interface{}) interface{} {
 func NewService(dbServer *DBServer) *ServOVSDB {
 	return &ServOVSDB{dbServer: dbServer}
 }
+
+// TODO move to utilities
+func arrayToMap(input []interface{}) map[interface{}]bool {
+	ret := map[interface{}]bool{}
+	for _, str := range input {
+		ret[str] = true
+	}
+	return ret
+}
+
+func (s *ServOVSDB) getMonitoredData(dataBase string, conditions map[string][]ovsjson.MonitorCondRequest) (map[string]map[string]ovsjson.RowUpdate2, error) {
+
+	returnData := map[string]map[string]ovsjson.RowUpdate2{}
+	for tableName, mcrs := range conditions {
+		resp, err := s.dbServer.GetData("ovsdb/"+dataBase+"/"+tableName, false)
+		if err != nil {
+			return nil, err
+		}
+		if dataBase == "_Server" && tableName == "Database" {
+			if len(mcrs) > 1 {
+				fmt.Printf("!!!!! MCR is not a singe %v\n", mcrs)
+			}
+			d1 := map[string]ovsjson.RowUpdate2{}
+			for _, v := range resp.Kvs {
+				//	keys := strings.Split(string(v.Key), "/")
+				fmt.Printf("key = %v \n", string(v.Key))
+				data := map[string]interface{}{}
+				json.Unmarshal(v.Value, &data)
+				uuidSet := data["uuid"]
+				uuid := uuidSet.([]interface{})[1]
+				if len(mcrs[0].Columns) == 0 {
+					delete(data, "uuid")
+					delete(data, "version")
+				} else {
+					columnsMap := arrayToMap(mcrs[0].Columns)
+					for column, _ := range data {
+						if _, ok := columnsMap[column]; !ok {
+							delete(data, column)
+						}
+					}
+				}
+				d1[uuid.(string)] = ovsjson.Initial{Initial: data}
+			}
+			returnData[tableName] = d1
+		} else {
+			// TODO
```

Comment:
more helpful is you explain what is needed in the TODO, at least for me it is unclear what is missing

Reply:
done